xonsh/built_ins.py

```diff
@@ -540,7 +539,6 @@ def prep_preexec_fn(self, kwargs, pipeline_group=None):
         else:
             def xonsh_preexec_fn():
                 """Preexec function bound to a pipeline group."""
-                os.setpgid(0, pipeline_group)
```

Comment:
I think we want to keep this around for Linux.  Also maybe add a comment that we aren't sure why it doesn't work on Mac.